include/seastar/net/tcp.hh

```diff
@@ -2089,8 +2087,14 @@ tcp_seq tcp<InetTraits>::tcb::get_isn() {
     hash[0] = _local_ip.ip;
     hash[1] = _foreign_ip.ip;
     hash[2] = (_local_port << 16) + _foreign_port;
-    hash[3] = _isn_secret.key[15];
-    CryptoPP::Weak::MD5::Transform(hash, _isn_secret.key);
+    gnutls_hash_hd_t md5_hash_handle;
+    // GnuTLS digests do not init at all, so this should never fail.
+    gnutls_hash_init(&md5_hash_handle, GNUTLS_DIG_MD5);
+    gnutls_hash(md5_hash_handle, hash, 3 * sizeof(hash[0]));
+    gnutls_hash(md5_hash_handle, _isn_secret.key, sizeof(_isn_secret.key));
```

Comment:
Isn't this changing the algorithm slightly? In the old code, we hashed key[15] twice. Probably it doesn't matter. @asias

Reply:
agreed, this does change the existing algoritm. see https://cryptopp.com/docs/ref/md5_8cpp_source.html . i will update the commit message to reflect this change. but i don't think it matters.

Reply:
@asias please ack